silverback/cluster/types.py

```diff
@@ -0,0 +1,281 @@
+import enum
+import math
+import uuid
+from datetime import datetime
+from typing import Annotated
+
+from pydantic import BaseModel, Field, computed_field, field_validator
+
+
+class WorkspaceInfo(BaseModel):
+    id: uuid.UUID
+    owner_id: uuid.UUID
+    name: str
+    slug: str
+
+
+class ClusterConfiguration(BaseModel):
+    """Configuration of the cluster (represented as 16 byte value)"""
+
+    # NOTE: This configuration must be encode-able to a uint64 value for db storage
+    #       and on-chain processing through ApePay
+
+    # NOTE: All defaults should be the minimal end of the scale,
+    #       so that `__or__` works right
+
+    # Version byte (Byte 0)
+    # NOTE: Just in-case we change this after release
+    version: int = 1
+
+    # Bot Worker Configuration (Bytes 1-2)
+    cpu: Annotated[int, Field(ge=0, le=16)] = 0  # 0.25 vCPU
+    """Allocated vCPUs per bot: 0.25 vCPU (0) to 16 vCPU (6)"""
+
+    memory: Annotated[int, Field(ge=0, le=120)] = 0  # 512 MiB
+    """Total memory per bot (in GB)"""
```

Comment:
question: If this is in GB, why does the comment say 512MB? And do we want to allow 120GB of memory for bots?

Reply:
I got this from AWS docs, that's the maximum they currently have.

Everything is in GB except 0, which is a special value for 512 MiB

Reply:
Again, got what from AWS docs? My question is about your comment and our upper bound.

Reply:
yes the 512 MiB is weird, but useful. I can explain it better that it is a special case.

120 GB does seem crazy to me, but it's priced into the model and some people might do weird things with AI that take a lot of memory, idk